structopt-derive/src/attrs.rs

```diff
@@ -371,14 +371,49 @@ impl Attrs {
         let comment_parts: Vec<_> = attrs
             .iter()
             .filter(|attr| attr.path.is_ident("doc"))
-            .filter_map(|attr| {
-                if let Ok(NameValue(MetaNameValue { lit: Str(s), .. })) = attr.parse_meta() {
-                    Some(s.value())
-                } else {
-                    // non #[doc = "..."] attributes are not our concern
-                    // we leave them for rustc to handle
-                    None
+            .filter_map(|attr| match attr.parse_meta() {
+                Ok(NameValue(MetaNameValue { lit: Str(s), .. })) => Some(s.value()),
+                Ok(NameValue(_)) => None,
+                Ok(Path(_)) => None,
+                // external-doc
+                // eg. #[doc(include = "../README.md")]
+                #[cfg(feature = "external_doc")]
+                Ok(List(syn::MetaList { nested, .. })) => {
+                    let content: String = nested
+                        .iter()
+                        .filter_map(|nested_meta| match nested_meta {
+                            syn::NestedMeta::Meta(syn::Meta::NameValue(nv)) => {
+                                if nv.path.is_ident("include") {
+                                    match &nv.lit {
+                                        Str(s) => {
+                                            let path = std::path::PathBuf::from("src")
+                                                .join(s.value())
+                                                .canonicalize()
+                                                .unwrap();
+                                            Some(std::fs::read_to_string(&path).unwrap())
```

Comment:
Please use `abort!` instead of `unwrap`s here

Reply:
Changed it! Sorry for not checking the library culture; and thanks for indicating it (it is very cool)